Sourcery/Models/TypeName.swift

```diff
@@ -82,11 +82,21 @@ public final class TypeName: NSObject, AutoCoding, AutoEquatable, AutoDiffable,
                 self.unwrappedTypeName = name
             }
         }
+        
+        if let characters = self.unwrappedTypeName.characters.split(separator: "<").first {
+            self.baseTypeName = String(characters)
+        } else {
+            self.baseTypeName = ""
+            
+        }
     }
 
     /// Type name used in declaration
     public let name: String
-
+    
+    /// Type name without the generics
+    public var baseTypeName: String
```

Comment:
all the types have `typeName: TypeName` property, so generic info will be accessed like this `type.typeName.generic.baseTypeName` or something like that for any type, even for `Protocol`